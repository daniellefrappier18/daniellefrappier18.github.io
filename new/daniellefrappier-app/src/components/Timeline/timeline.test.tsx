import React from 'react';
import { Timeline } from './timeline.component';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

const jobs = [
    {
      title: "Senior Systems Design Engineer",
      date: "/ September 2021 - Current",
      location: "Monster",
      description: (
        <>
          <li>Team lead for Randstad Brand Design System. As Team Lead I was responsible for mentoring junior engineers, establishing code standards and best practices and making decisions on the architecture of the design system.</li>
          <li>Built React component library with TypeScript Styled Components, wrote unit tests using Jest and React Testing Library and documented components in Storybook.</li>
          <li>Provided support for monster.com homepage built using Gatsby.</li>
        </>)
    }
  ];
describe('Timeline component', () => {
    // test component with props
    it('Timeline renders correctly Props', () => {
      const tree = renderer
        .create(
          <Timeline
           title='Work Experience'
           jobs={jobs}
          />,
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
    afterEach(cleanup);
  });