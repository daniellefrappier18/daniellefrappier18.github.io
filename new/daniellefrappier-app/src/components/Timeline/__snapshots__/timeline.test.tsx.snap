// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline component Timeline renders correctly Props 1`] = `
<div
  className="timelinestyle__TimelineComponent-sc-e6g3ra-0 cCXEpk"
>
  <h2
    className="timelinestyle__TimelineTitle-sc-e6g3ra-1 iCuyKS"
  >
    Work Experience
  </h2>
  <div
    className="timelinestyle__Item-sc-e6g3ra-2 gRqQGD"
  >
    <div
      className="timelinestyle__ItemBullet-sc-e6g3ra-3 eoyuly"
    />
    <div
      className="timelinestyle__ItemContent-sc-e6g3ra-4 jyrCNP"
    >
      <div
        className="timelinestyle__ItemTitleContainer-sc-e6g3ra-5 AfuSy"
      >
        <h3
          className="timelinestyle__ItemTitle-sc-e6g3ra-6 jzrHOl"
        >
          Senior Systems Design Engineer
          <span
            className="timelinestyle__ItemSubtitle-sc-e6g3ra-7 cJcfTz"
          >
            / September 2021 - Current
             
            <br />
             
            Monster
          </span>
        </h3>
      </div>
      <ul
        className="timelinestyle__ItemText-sc-e6g3ra-8 clbOGA"
      >
        <li>
          Team lead for Randstad Brand Design System. As Team Lead I was responsible for mentoring junior engineers, establishing code standards and best practices and making decisions on the architecture of the design system.
        </li>
        <li>
          Built React component library with TypeScript Styled Components, wrote unit tests using Jest and React Testing Library and documented components in Storybook.
        </li>
        <li>
          Provided support for monster.com homepage built using Gatsby.
        </li>
      </ul>
    </div>
  </div>
</div>
`;
